satellite/overlay/selection_test.go

```diff
@@ -301,14 +301,61 @@ func TestNodeSelectionGracefulExit(t *testing.T) {
 	})
 }
 
+func TestFindStorageNodesDistinctIPs(t *testing.T) {
+	if runtime.GOOS == "darwin" {
+		t.Skip("Test does not work with macOS")
+	}
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 5, UplinkCount: 1,
+		Reconfigure: testplanet.Reconfigure{
+			// will create 3 storage nodes with same IP; 2 will have unique
+			UniqueIPCount: 2,
+		},
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		satellite := planet.Satellites[0]
+
+		var excludedNodes storj.NodeIDList
+		addrs := make(map[storj.NodeID]struct{})
+		for _, node := range planet.StorageNodes {
+			if _, ok := addrs[node.ID()]; ok {
```

Comment:
shouldn't the condition be `!ok`? I don't see how anything can be added to `addrs` otherwise

Reply:
Also `node.ID()` will be unique for every storagenode. Shouldn't `addrs` be a `map[string]struct{}` with IP being the key instead?

Reply:
oh shoot idk what was happening here!